test/lit_tests/binop_const_lift_computation.mlir

```diff
@@ -0,0 +1,105 @@
+// RUN: enzymexlamlir-opt --enzyme-hlo-opt %s | FileCheck %s
+
+// (x * 2) / 4 == x * (2 / 4)
```

Comment:
Just for the sake of testing out that the transform dialect stuff still works here, can we also add a test adding it via transform ops (and similarly add the presumably autogenerated patterns to https://github.com/EnzymeAD/Enzyme-JAX/blob/309095e2cd78adc5fca96e4748cfa67cbf55c17b/src/enzyme_ad/jax/primitives.py#L231

Reply:
cc @ftynse re transform

Reply:
Updated the old `binop_const_simplify` to use these new patterns. Should I add a test file, or would that just be testing mlir impl details?

Reply:
That’s fine for now. Longer term (to make it easier to migrate existing patterns to tablegen), we should probably have the tablegen patterns autogenerate the transform names too.

Reply:
> cc @ftynse re transform

It's easy an op per pattern, just copy paste the existing one and change the name. Not sure it is needed here though, the patterns look generally useful.

Reply:
Yeah just having all the patterns in transform is just immensely useful for debugging issues